File: GFG/Arrays/freqinsorted.cpp
#include<iostream>
#include<vector>
using namespace std;

void frequency(vector<int> &arr){
    int freq=1;
    int n= arr.size();
    int i =1;
    while (i<n){
        while(i<n and arr[i]==arr[i-1]){
            freq++;
            i++;
        }
        cout<<arr[i-1]<<" "<<freq<<endl;
        freq=1;
        i++;
    }
    if (n==1 || arr[n-1]!=arr[n-2]){
        cout<<arr[n-1]<<" "<<"1"<<endl;
    }
}
int main (){
    vector<int> arr= {5,10,10,10,20,20,30,40,40,40,40,50,50,60};
    frequency(arr);
    return 0;
}

File: GFG/Arrays/freqinsorted.py
def frequency(arr):
    freq=1
    j=1
    n = len(arr)
    while(j<n):
        while( j<n and arr[j]==arr[j-1]):
            freq+=1
            j+=1
        print(arr[j-1]," ",freq)
        freq=1
        j+=1
    if(n==1 or arr[n-1]!=arr[n-2] ):
        print(arr[n-1],"  1")
arr=[5,10,10,10,20,20,30,30,30,30,40,40,50,60]
frequency(arr)
        